SVProfilePictureViewer/PictureViewerForm.cs

```diff
@@ -139,6 +145,23 @@ enum MaskType {
           }
           return result;
         }
+        //the most similiar to the IRL one
+        case BlendType.Dynamic: {
```

Comment:
As I said it's a variation of the algorithm I presented on the forum since it worked in underwhelming manner, but it could also have been an incorrect c# translation by me

Reply:
Based on what is happening in the current code where m1 and m2 are being used in the pixel calculation without the alpha actually being taken into account they should be replaced with a single copy of  the light component.

Reply:
Sorry I'm afraid I didn't understand correctly.
M1 and M2 take in consideration alpha since the two masks have been applied on the light image. The result is an average among 2 images derived from light image and the single dark image. Thus elements from the dark image are preserved in the empty spots where both M1 and M2 had an alpha of 255, while where one or both the masks have visible pixels they lighten and smooth the corresponding pixels on the dark image.

Again this is a revisitation of the algorithm I described in the forum (a sort of hacky way to correct lighting in the image). Furthermore the previous algorithm of the average between light and dark image was indeed a hack, because it corresponds to the alpha blending algorithm I described in the forum, with a value of exactly 0.5.

The direct translation of my recent Python code didn't work as expected in c#, but it could have been an erroneous translation by my side. If you don't want to take account of this variation I could always commit the direct translation.

Reply:
While you are setting the alpha values in m1 and m2 when using the Color objects from GetPixel on the images, you are using the RGB values that are unaffected by the alpha.

Reply:
It most likely is because alpha is an int when it should be a float or double so when dividing by 255 it is rounding to zero since C# truncates its where Python would return a floating point number